caluma/caluma_form/structure.py

```diff
@@ -144,18 +150,18 @@ def __init__(self, document, form, question=None, parent=None):
         self.question = question
         self._fields = None
         self._sub_forms = None
-        self._main_case_form = "NOTSET"
+        self._case = "NOTSET"
 
     @property
-    def main_case_form(self):
-        if self._main_case_form == "NOTSET":
+    def case(self):
+        if self._case == "NOTSET":
             try:
-                self._main_case_form = (
-                    self.document.family.work_item.case.family.document.form.slug
-                )
+                case = self.document.family.work_item.case
+                parent = case.parent_work_item.case if case.parent_work_item else None
+                self._case = Case(case, parent=Case(parent))
```

Comment:
@winged For some reason, the `parent` weakref was dead when trying to read it, which made some of the tests fail. I removed the weakref abstraction just to get the tests green, but I'm sure that's not what we actually want :-)

Reply:
Yeah we need to find another solution for that. Making `parent` a regular ref turns the structure into a reference loop that the GC can't collect, causing memory leaks.

For the `case` here though, I think we could get away with it, as the `Case` element does not present access to the form, so is not circular

Reply:
Fixed in pair programming session, thanks again @winged